src/app/drive/components/ShareInviteDialog/ShareInviteDialog.tsx

```diff
@@ -61,12 +67,23 @@ const ShareInviteDialog = (props: ShareInviteDialog): JSX.Element => {
     }
   };
 
-  const onEditRole = () => {
-    //   Edit user role
+  const onEditRole = (value: PrivateSharingRole['role'], user: UsersToInvite) => {
+    const newUserToInvite = usersToInvite.map((userToInvite) => {
+      if (user.email === userToInvite.email) {
+        return { ...userToInvite, userRole: value };
+      }
+      return userToInvite;
+    });
+    setUsersToInvite(newUserToInvite);
   };
 
   const onInvite = () => {
-    // Ivite added users
+    usersToInvite.forEach((user) => {
+      const userRoleId = props.roles.find((role) => role.role === user.userRole)?.id;
```

Comment:
Shouldn't this be `user.role`?

Reply:
No, it is the format of the backend `roles` object. `role.role` is the name of the role

Reply:
Okay, try to abstract your data structures from the backend, we should not depend on how the backend chooses to expose things, or any change there will break the entire frontend. Instead of that, I recommend a layer in-between which may assert that the expected data comes from the backend and transforms that to the preferable data structure. That also will make testing easier & faster

Reply:
Totally agree, I'm going to extract the call to get the roles from the ShareDialog view and add the layer already in this PR and so it is already prepared, as it is not definitive I was putting together the calls to extract it all at once once once we had them all.